src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateToChange = { ...state };
+  let currentState = { ...stateToChange };
+  const iterationsOfTransform = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'addProperties':
+        Object.assign(stateToChange, actions[i].extraData);
+        break;
+
+      case 'removeProperties':
+        for (const keyToRemove of actions[i].keysToRemove) {
+          delete stateToChange[keyToRemove];
+        }
+        break;
+
+      case 'clear':
+        for (const keyToClear in stateToChange) {
+          delete stateToChange[keyToClear];
+        }
+        break;
+
+      default:
+        return [{}];
```

Comment:
you don't need this return here, just default: nextState = { ...currentState }; or empty;